src/scripts/main.js

```diff
@@ -1,3 +1,45 @@
 'use strict';
 
 // write code here
+const headline = document.querySelector('thead');
+
+function getCleanedValue(row, index) {
+  const cells = row.querySelectorAll('td')[index];
+
+  if (!cells) {
+    return '';
+  }
+
+  const cleanedCell = cells.textContent.trim().replace(/[$,]/g, '');
+
+  return isNaN(cleanedCell) ? cleanedCell : parseInt(cleanedCell);
+}
+
+headline.addEventListener('click', (e) => {
+  const clickedElement = e.target.closest('th');
+
+  if (!clickedElement) {
+    return;
+  }
+
+  const columnIndex = clickedElement.cellIndex;
+  const tableRows = [...document.querySelectorAll('tbody tr')];
+  const tbody = document.querySelector('tbody');
+
+  tableRows.sort((row1, row2) => {
+    const value1 = getCleanedValue(row1, columnIndex);
+    const value2 = getCleanedValue(row2, columnIndex);
+
+    if (typeof value1 === 'number' && typeof value2 === 'number') {
+      return value1 - value2;
+    }
+
+    if (typeof value1 === 'string' && typeof value2 === 'string') {
+      return value1.localeCompare(value2);
+    }
+  });
+
+  tableRows.forEach((rowData) => {
+    tbody.appendChild(rowData);
+  });
```

Comment:
Code logic issue: You're appending the sorted rows to the `tbody` element, but you're not removing the old rows first. This will result in duplicate rows in the table. You should clear the `tbody` before appending the sorted rows.